cibuildwheel/options.py

```diff
@@ -118,13 +118,16 @@ def architectures(self) -> set[Architecture]:
         return self.globals.architectures
 
 
-Setting = Union[Mapping[str, str], Sequence[str], str, int, bool]
+SettingLeaf = Union[str, int, bool]
+SettingTable = Mapping[str, Union[SettingLeaf, Sequence[SettingLeaf]]]
+SettingList = Sequence[SettingLeaf]
```

Comment:
I guess the intent is really the following ?
```suggestion
SettingList = Sequence[SettingLeaf]
SettingTable = Mapping[str, Union[SettingLeaf, SettingList]]
```

Reply:
Yeah, I think so...